### PR TITLE
Fix/various

### PR DESCRIPTION
**Pull request by martinfalk**

### CODE REVIEW

File: include/inviwo/core/util/raiiutils.h [hunk 1]
```diff
@@ -39,11 +39,18 @@ namespace util {
 
 class IVW_CORE_API KeepTrueWhileInScope {
 public:
-    KeepTrueWhileInScope(bool* b) : variable_(b) { if(variable_) (*variable_) = true; }
-    ~KeepTrueWhileInScope() { if(variable_) (*variable_) = false; }
```
Comment (petersteneteg) [inline include/inviwo/core/util/raiiutils.h]: Are we sure we don’t depend on this behavior anywhere?

**Review by petersteneteg — APPROVED** [review]

File: include/inviwo/core/util/raiiutils.h (hunk 1 above)
Comment (martinfalk) [reply include/inviwo/core/util/raiiutils.h]: Cannot guarantee that. People might have a different interpretation of what `KeepTrueWhileInScope` should be doing (see comment by @r-englund in #204)

Comment (martinfalk) [reply include/inviwo/core/util/raiiutils.h]: Still wondering though why the canvas was shown in the first place. Maybe something within Qt changed slightly and the visibility changes now trigger signals. Who knows ...

Comment (petersteneteg) [reply include/inviwo/core/util/raiiutils.h]: I would be very careful with this, might break serialisation in hard to notice ways...

Comment (martinfalk) [reply include/inviwo/core/util/raiiutils.h]: And it might be incorrect in many others as well.

**Pull request merged**